install/kubernetes/cilium/templates/cilium-agent/ipsec-secret.yaml

```
@@ -0,0 +1,16 @@
+{{- if and .Values.encryption.enabled (eq .Values.encryption.type "ipsec") }}
+{{- if .Values.encryption.ipsec.createSecret }}
+{{- $keys := .Values.encryption.ipsec.keyData | required "must set encryption.ipsec.keyData when creating secret" | quote }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.encryption.ipsec.secretName | default .Values.encryption.secretName }}
+  namespace: {{ .Release.Namespace }}
+stringData:
+  {{- if and .Values.encryption.ipsec.mountPath .Values.encryption.ipsec.keyFile }}
```

is the check for mountPath here required? I am thinking about something like below

```suggestion
{{- $secretName := .Values.encryption.ipsec.secretName | default .Values.encryption.secretName }}
{{- $keyFile := .Values.encryption.ipsec.keyFile | default .Values.encryption.keyFile }}
apiVersion: v1
kind: Secret
metadata:
  name: {{ $secretName }}
  namespace: {{ .Release.Namespace }}
stringData:
  {{ $keyFile }}: {{ $keys }}
{{- end }}
{{- end }}
```

---

Hmm, yeah, I think that would be better too. I got this check from the configmap – do you think it should be changed as well? 

https://github.com/cilium/cilium/blob/ed14af8b4c514deef12dcd1a12678de5110352b1/install/kubernetes/cilium/templates/cilium-configmap.yaml#L444-L448

---

I guess it matters more not to set the `ipsec-key-file` value unless it's mounted there too, so +1 to change it here regardless.

---

yes, mountPath is only used in when secret is used. The changes we are doing here are to create the secret.